purchases/src/main/kotlin/com/revenuecat/purchases/common/LocaleProvider.kt

```diff
@@ -7,6 +7,29 @@ internal interface LocaleProvider {
 }
 
 internal class DefaultLocaleProvider : LocaleProvider {
+    private var orchestratorProvider: (() -> com.revenuecat.purchases.PurchasesOrchestrator?)? = null
```

Comment:
Hmm TBH, I'm not liking adding a dependency to the purchases orchestrator from the locale provider... Could we create a separate holder like `LocaleOverrideData` which can be passed here? If we make the setter in `Purchases` a static method that sets this data statically, this could be even easier.